exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/transactions/TransferTxTest.java

```diff
@@ -142,45 +185,47 @@ void executeTransfer() throws CloseFailuresException {
 
   @Test
   @RequiresNativeLibrary
-  void executeTransferToTheSameWallet() throws CloseFailuresException {
+  void executeNoSuchFromWallet() throws CloseFailuresException {
     try (Database db = MemoryDb.newInstance();
          Cleaner cleaner = new Cleaner()) {
       Fork view = db.createFork(cleaner);
+      // Create the receiver wallet with the given initial balance
+      long initialBalance = 50L;
+      createWallet(view, toKey, initialBalance);
 
-      long initialBalance = 100L;
-      createWallet(view, fromKey, initialBalance);
-
-      // Create and execute the transaction
       long seed = 1L;
-      long transferSum = 40L;
-      TransferTx tx = withMockMessage(seed, fromKey, fromKey, transferSum);
-      tx.execute(view);
 
-      // Check that the balance of the wallet remains the same
+      long transferValue = 50L;
+      TransferTx tx = withMockMessage(seed, fromKey, toKey, transferValue);
+      // Execute the transaction that attempts to transfer from an unknown wallet
+      TransactionExecutionException e = assertThrows(
+          TransactionExecutionException.class, () -> tx.execute(view));
+      assertThat(e.getErrorCode(), equalTo(UNKNOWN_SENDER.errorCode));
+
+      // Check that balance of fromKey is unchanged
       CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      ProofMapIndexProxy<PublicKey, Wallet> wallets = schema.wallets();
-      assertThat(wallets.get(fromKey).getBalance(), equalTo(initialBalance));
+      MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+      assertThat(wallets.get(toKey).getBalance(), equalTo(initialBalance));
     }
   }
 
   @Test
   @RequiresNativeLibrary
-  void executeNoSuchFromWallet() throws CloseFailuresException {
+  void executeNoSuchToWallet() throws CloseFailuresException {
     try (Database db = MemoryDb.newInstance();
          Cleaner cleaner = new Cleaner()) {
       Fork view = db.createFork(cleaner);
-      // Create source wallet with the given initial balance
-      long initialBalance = 50L;
+      // Create and execute the transaction that attempts to transfer to unknown wallet
```

Comment:
probably this commend should be a few lines below

Reply:
:+1: moved in other tests too